backend/tracim_backend/lib/core/content.py

```diff
@@ -536,29 +541,34 @@ def create(self, content_type_slug: str, workspace: Workspace, parent: Content=N
                         content_id=workspace.workspace_id,
                     )
                 )
-        if filename:
-            label = os.path.splitext(filename)[0]
-        if label:
-            self._is_content_label_free_or_raise(
-                label,
+        content = None
+        if filename or label:
+            if label:
+                file_extension = ''
+                if content_type.slug in (
+                        CONTENT_TYPES.Page.slug,
+                        CONTENT_TYPES.Thread.slug,
+                ):
+                    file_extension = '.html'
+                filename = '{}{}'.format(label, file_extension)
+            self._is_filename_free_or_raise(
+                filename,
                 workspace,
                 parent,
             )
-
-        content = Content()
-        if filename:
+            content = Content()
             # INFO - G.M - 2018-07-04 - File_name setting automatically
             # set label and file_extension
-            content.file_name = label
-        elif label:
-            content.label = label
+            content.file_name = filename
         else:
             if content_type_slug == CONTENT_TYPES.Comment.slug:
+                content = Content()
                 # INFO - G.M - 2018-07-16 - Default label for comments is
                 # empty string.
                 content.label = ''
             else:
-                raise EmptyLabelNotAllowed('Content of this type should have a valid label')  # nopep8
+                raise EmptyLabelNotAllowed(
+                    'Content of this type should have a valid label')  # nopep8
```

Comment:
replace this type by the real content type. It's better for the user or developer to understand the exception (and to detect a wrong content type)